scripts/Muon/GUI/Common/contexts/plotting_context.py

```diff
@@ -0,0 +1,156 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from Muon.GUI.Common.contexts.plot_edit_context import PlotEditContext
+from typing import List
+from enum import Enum
+
+
+PlotModeStrings = ['Plot Data', 'Plot Fits']
+
+
+class PlotMode(Enum):
+    Data = 0
+    Fitting = 1
+
+    def __str__(self):
+        return PlotModeStrings[self.value]
+
+
+class PlottingContext(object):
+    def __init__(self):
+        self._subplots = {}
+        self._default_xlim = [-0.1 ,0.1]
+        self._default_ylim = [-10, 10]
+        self._xlim_all = self._default_xlim
+        self._ylim_all = self._default_ylim
```

Comment:
self._xlim_all and self._ylim_all are updated in a lot of places but are is not used to do anything

Reply:
https://github.com/mantidproject/mantid/pull/31148/files#diff-4fd12c811d5cd8d863b95144eb3973a5ac018e26c77da7919fc16e64f81cc0deR179

Reply:
https://github.com/mantidproject/mantid/pull/31148/files#diff-4fd12c811d5cd8d863b95144eb3973a5ac018e26c77da7919fc16e64f81cc0deR179